query/parse.go

```diff
@@ -154,6 +155,13 @@ func parseExpr(in []byte) (Q, int, error) {
 	case tokBranch:
 		expr = &Branch{Pattern: text}
 	case tokText, tokRegex:
+		if !enableRegex {
```

Comment:
instead of esacping the regex and calling `RegexpQuery`, can we instead just set the `expr` to a `Substring` object (which is what `RegexpQuery` does internally in case it's not a regex).
So sth like:
``` go
if !enableRegex {
    expr = &Substring{
          Pattern:  text,
          FileName: file,
          Content:  content,
    }
    break
}
```

That way we would avoid escaping back and forth on regex?
Might be worth a try if you didn't try it yet.

Reply:
We get an error in this case "query: missing close paren, got token" for this query `Scan(src`

Reply:
Query needs to be `"Scan(src"` for the above example otherwise zoekt thinks that closing bracket is absent and fails the query.

Reply:
What's the error you get?

Reply:
No errors with `"Scan(src"`. It gives 0 results.

Reply:
weird, anyway don't want to block this. can be improved later if needed.